firestore/google/cloud/firestore_v1beta1/_helpers.py

```diff
@@ -802,37 +804,51 @@ def remove_server_timestamp(document_data):
        }
 
     Args:
-        document_data (dict): Property names and values to use for
-            sending a change to a document.
+        document_data (dict):
+            Property names and values to use for sending a change to
+            a document.
+
+        split_on_dots (bool):
+            Whether to split the property names on dots at the top level
+            (for updates only).
 
     Returns:
         Tuple[List[str, ...], Dict[str, Any]]: A two-tuple of
 
-        * A list of all field paths that use the server timestamp sentinel
+        * A list of all transform paths that use the server timestamp sentinel
         * The remaining keys in ``document_data`` after removing the
           server timestamp sentinels
     """
     field_paths = []
+    transform_paths = []
     actual_data = {}
     for field_name, value in six.iteritems(document_data):
         if isinstance(value, dict):
-            sub_field_paths, sub_data = remove_server_timestamp(value)
-            field_paths.extend(
-                get_field_path([field_name, sub_path])
-                for sub_path in sub_field_paths
-            )
+            sub_transform_paths, sub_data, sub_field_paths = (
+                process_server_timestamp(value, False))
+            for sub_path in sub_transform_paths:
+                field_path = FieldPath.from_string(field_name)
+                field_path.parts = field_path.parts + sub_path.parts
+                transform_paths.extend([field_path])
```

Comment:
Nit: Change all the naming in here to use 'transform' somewhere. That way, it is easy to tell the difference between this block and the one just below.